### PR TITLE
Implement vehicle refit game command

### PR DESCRIPTION
**Pull request by AaronVanGeffen**

### CODE REVIEW

File: src/OpenLoco/src/GameCommands/VehicleRefit.cpp [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+#include "Ui/WindowManager.h"
+#include "Vehicles/Vehicle.h"
+#include <OpenLoco/Interop/Interop.hpp>
+#include <OpenLoco/Utility/Numeric.hpp>
+
+using namespace OpenLoco::Interop;
+
+namespace OpenLoco::GameCommands
+{
+    // 0x004B0B50
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/GameCommands/VehicleRefit.cpp:17]:
```suggestion
    // 0x0042F6DB
```

File: src/OpenLoco/src/Vehicles/VehicleBody.cpp [hunk 2]
```diff
@@ -1327,4 +1328,13 @@ namespace OpenLoco::Vehicles
             invalidateSprite();
         }
     }
+
+    // 0x0042F6B6
+    // TODO: move this?
+    uint32_t getNumUnitsForCargo(uint32_t maxPrimaryCargo, uint8_t primaryCargoId, uint8_t newCargoId)
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Vehicles/VehicleBody.cpp:1334]: Not sure if this is the right place for this function. I needed to move it out of the vehicle window, though -- it is called in three more locations.

File: src/OpenLoco/src/GameCommands/VehicleRefit.cpp [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+using namespace OpenLoco::Interop;
+
+namespace OpenLoco::GameCommands
+{
+    // 0x0042F6DB
+    static uint32_t vehicleRefit(const VehicleRefitArgs& args, const uint8_t flags)
+    {
+        setExpenditureType(ExpenditureType::TrainRunningCosts);
+
+        try
```
Comment (LeftofZen) [inline src/OpenLoco/src/GameCommands/VehicleRefit.cpp:22]: is this standard in openloco? what are we expecting to throw an exception here?

File: src/OpenLoco/src/Vehicles/VehicleBody.cpp (hunk 2 above)
Comment (LeftofZen) [reply src/OpenLoco/src/Vehicles/VehicleBody.cpp:1334]: shouldn't we just keep the definition here and keep the body in the cpp?

File: src/OpenLoco/src/GameCommands/VehicleRefit.cpp (hunk 3 above)
Comment (AaronVanGeffen) [reply src/OpenLoco/src/GameCommands/VehicleRefit.cpp:22]: Creating a Vehicle can throw an exception, yeah.

**Review by LeftofZen — CHANGES_REQUESTED** [review]
playtest - the refit itself works, however in routing tab you are unable to set orders for the refitted cargo. for example: got a plane originally with passengers, refitted to mail:

image

in routing, can only select passengers

image

I tested this in current 23.05 and that does correctly show the routing orders for mail only, so this PR has an issue

**Comment by AaronVanGeffen** [comment]
@LeftofZen Thanks, should be fixed now. Could you retest? 🙂

File: src/OpenLoco/src/GameCommands/VehicleRefit.cpp [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+            auto* vehObj = ObjectManager::get<VehicleObject>(car.body->objectId);
+
+            if (!vehObj->hasFlags(VehicleObjectFlags::refittable))
+            {
+                setErrorText(StringIds::empty);
+                return 0;
+            }
+
+            auto carIter = train.cars.begin();
+            carIter++;
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/GameCommands/VehicleRefit.cpp:53]: This looks wrong. I'm not sure what you are doing here. Shouldn't it just be `car.body` from L44

File: src/OpenLoco/src/Vehicles/VehicleBody.cpp (hunk 2 above)
Comment (duncanspumpkin) [reply src/OpenLoco/src/Vehicles/VehicleBody.cpp:1334]: Its really a cargo function i guess. Not sure where would be best

Comment (AaronVanGeffen) [reply src/OpenLoco/src/Vehicles/VehicleBody.cpp:1334]: Maybe move it to be part of the `CargoObject`? Also not ideal, considering it's working with two objects... Perhaps we should just leave it here, for now?

**Comment by LeftofZen** [comment]
> @LeftofZen Thanks, should be fixed now. Could you retest? 🙂

Retested, routing orders look correct now, nice job!

**Review by LeftofZen — APPROVED** [review]

File: src/OpenLoco/src/GameCommands/VehicleRefit.cpp (hunk 4 above)
Comment (AaronVanGeffen) [reply src/OpenLoco/src/GameCommands/VehicleRefit.cpp]: Nice catch. Much cleaner this way, and works just as well.

File: src/OpenLoco/src/GameCommands/VehicleRefit.cpp [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+            }
+
+            if (train.cars.empty())
+            {
+                setErrorText(StringIds::empty);
+                return 0;
+            }
+
+            auto car = train.cars.firstCar;
+            auto* vehObj = ObjectManager::get<VehicleObject>(car.body->objectId);
```
[the remark this answers is not in the document]
Comment (AaronVanGeffen) [reply src/OpenLoco/src/GameCommands/VehicleRefit.cpp:44]: Just for my clarity, it should have read `car.front->objectId` if I had followed vanilla, correct? I'm fine either way.

File: src/OpenLoco/src/Vehicles/VehicleBody.cpp (hunk 2 above)
Comment (duncanspumpkin) [reply src/OpenLoco/src/Vehicles/VehicleBody.cpp:1334]: Yeah leave for now

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**